### DISCUSSION TITLE
Integration with  open source video players like (video.js or plyr)

### DISCUSSION THREAD

**Discussion by programmeddeath1** [Lumieducation/H5P-Nodejs-library#1961] [Q&A]
Hi ,
Thanks for Making a version of h5p in Node.js ,its really helpful , however the core  video player of h5p doesn't have many features that open source video players like (video.js/plyr) have , is there any plugin for these video players via h5p if not then can you please provide info on how to integrate it with these video players.

Thanks in Advance

**Comment by sr258** [comment]
As far as I'm aware there is no more powerful video player for H5P at the moment. You'd have to fork the existing video player (http://github.com/h5p/h5p-video) and add the functionality you want to it or completely replace it with a different library. You can then either upload it to your system with an increased patch number to replace the existing one or you can use a hook to replace the files of the regular library: https://docs.lumi.education/advanced-usage/customization#changing-javascript-and-css-files-of-individual-libraries

Uploaded a newer version is easier, but there's always the risk of a new minor version coming out, which would mean that newer content types don't use your player anymore, or a higher path version, which would mean that your customizations are replaced by the official version.

**Comment by programmeddeath1** [comment]
The current h5p video can also be extracted as an html file, it basically contains the script importing and running the libraries and the style. With the h5p content object in the html page. Can this be directly added to any existing html page (part of a website/app) and run as an interactive video?

**Reply by sr258** [reply]
Do you mean the h5p-html-exporter module? If so: Yes, you can also embed this into other websites, either through an iframe or by copying the JS code, the styles and the HTML to another page (might be a bit cumbersome as the file as very large).
